google-appengine/main.py

```diff
@@ -181,11 +183,11 @@ def importcsv():
 @authenticate
 def webconsole():
     if request.method == 'POST':
-        if request.form.has_key('student') and\
-            request.form.has_key('month') and \
-            request.form.has_key('day') and \
-            request.form.has_key('year') and \
-            request.form.has_key('action'):
+        if request.form.get('student') or \
```

Comment:
This seems to be a change in logic. Previously, the conditional was true only if all values were present. Now the conditional is true if any one value is present.

Reply:
The difference between `.has_key` and `.get` is that `.has_key` merely checks if the form has that field, while `.get` actually looks for that value in the form, returning None if there is nothing. Since `.has_key` got deprecated a while ago, so I simply updated the logic to make sure the input for a certain "action" worked out.